homeassistant/components/touchline/climate.py

```diff
@@ -95,8 +106,64 @@ def target_temperature(self):
         """Return the temperature we try to reach."""
         return self._target_temperature
 
+    @property
+    def current_operation_mode(self):
+        """Return the current operation mode."""
+        return self._current_operation_mode
+
+    @property
+    def preset_mode(self):
+        """Return the current preset mode."""
+        return self._preset_mode
+
+    @property
+    def preset_modes(self):
+        """Return available preset modes."""
+        return list(PRESET_MODES.keys())
+
+    def set_preset_mode(self, preset_mode):
+        """Set new target preset mode."""
+        mode = self.map_mode_hass_touchline(preset_mode)
+        program = self.map_program_hass_touchline(preset_mode)
+        self.unit.set_operation_mode(mode)
+        self.unit.set_week_program(program)
+
+    def set_hvac_mode(self, hvac_mode):
+        """Set new target hvac mode."""
+        self._current_operation_mode = HVAC_MODE_HEAT
+
     def set_temperature(self, **kwargs):
         """Set new target temperature."""
         if kwargs.get(ATTR_TEMPERATURE) is not None:
             self._target_temperature = kwargs.get(ATTR_TEMPERATURE)
         self.unit.set_target_temperature(self._target_temperature)
+
+    @staticmethod
+    def map_mode_hass_touchline(preset_mode):
+        """Map Home Assistant Preset Modes to Touchline Operation Modes."""
+        return PRESET_MODES[preset_mode]["mode"]
+
+    @staticmethod
+    def map_program_hass_touchline(preset_mode):
+        """Map Home Assistant Preset Modes to Touchline Program Modes."""
+        return PRESET_MODES[preset_mode]["program"]
```

Comment:
These methods are actually only used once and contain only a single line. Why not use them directly in place?

Reply:
Good point - fixed now.